sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -71,21 +71,26 @@ object JoinHint {
 case class HintInfo(strategy: Option[JoinStrategyHint] = None) {
 
   /**
-   * Combine two [[HintInfo]]s into one [[HintInfo]], in which the new strategy will the strategy
-   * in this [[HintInfo]] if defined, otherwise the strategy in the other [[HintInfo]].
+   * Combine this [[HintInfo]] with another [[HintInfo]] and return the new [[HintInfo]].
+   * @param other the other [[HintInfo]]
+   * @param hintOverriddenCallback a callback to notify if any [[HintInfo]] has been overridden
```

Comment:
if we create a hint merging strategy framework, I think it will not be an arbitrary callback. Shall we make it simple now and leave it for future design? Then we can just log message inside this method.